src/Module/Contact.php

```diff
@@ -355,54 +358,45 @@ public static function content(array $parameters = [], $update = 0)
 				throw new NotFoundException(DI::l10n()->t('Contact not found'));
 			}
 
-			if ($cmd === 'update' && ($orig_record['uid'] != 0)) {
-				self::updateContactFromPoll($contact_id);
-				DI::baseUrl()->redirect('contact/' . $contact_id);
+			$cdata = Model\Contact::getPublicAndUserContactID($orig_record['id'], local_user());
+
+			if ($cmd === 'update' && $cdata['user']) {
+				self::updateContactFromPoll($cdata['user']);
+				DI::baseUrl()->redirect('contact/' . $cdata['public']);
 				// NOTREACHED
 			}
 
-			if ($cmd === 'updateprofile') {
-				self::updateContactFromProbe($contact_id);
-				DI::baseUrl()->redirect('contact/' . $contact_id);
+			if ($cmd === 'updateprofile' && $cdata['user']) {
+				self::updateContactFromProbe($cdata['user']);
+				DI::baseUrl()->redirect('contact/' . $cdata['public']);
 				// NOTREACHED
 			}
 
 			if ($cmd === 'block') {
-				if (public_contact() === $contact_id) {
+				if (public_contact() === $cdata['public']) {
```

Comment:
Shouldn't we add this check to the batch operations as well?

Reply:
We can but it's less important, the batch operation is a POST request which makes it impossible to craft a malicious link to make someone block themselves or drop their own contact.

Reply:
But we should prevent that someone deletes their own contact by mistake, shouldn't we?

Reply:
Yes we should, but it isn't that pressing, we don't generate a POST form where your own contact is featured, and #10672 should prevent someone to craft a cross-posting form among other nastiness.